spark/src/test/scala/io/delta/sharing/spark/DeltaSharingSourceCDFSuite.scala

```diff
@@ -0,0 +1,460 @@
+
+  lazy val deltaLog = RemoteDeltaLog(cdfTablePath)
+
+  val streamingTimeout = 120.seconds
+
+  def getSource(parameters: Map[String, String]): DeltaSharingSource = {
+    val options = new DeltaSharingOptions(parameters ++ Map("readChangeFeed" -> "true"))
+    DeltaSharingSource(SparkSession.active, deltaLog, options)
+  }
+
+  def withStreamReaderAtVersion(
+      path: String = cdfTablePath,
+      startingVersion: String = "0"): DataStreamReader = {
+    spark.readStream.format("deltaSharing").option("path", path)
+      .option("startingVersion", startingVersion)
+      .option("ignoreDeletes", "true")
+      .option("ignoreChanges", "true")
+      .option("readChangeFeed", "true")
+  }
+
+  /**
+   * Test latestOffset for Stream CDF
+   */
+  integrationTest("DeltaSharingSource.latestOffset - startingVersion 0") {
+    val source = getSource(Map(
+      "ignoreChanges" -> "true",
+      "ignoreDeletes" -> "true",
+      "startingVersion" -> "0"
+    ))
+    val latestOffset = source.latestOffset(null, source.getDefaultReadLimit)
+
+    assert(latestOffset.isInstanceOf[DeltaSharingSourceOffset])
+    val offset = latestOffset.asInstanceOf[DeltaSharingSourceOffset]
+    assert(offset.sourceVersion == 1)
+    assert(offset.tableId == deltaLog.snapshot(Some(0)).metadata.id)
+    assert(offset.tableVersion == 5)
+    assert(offset.index == -1)
+    assert(!offset.isStartingVersion)
+  }
+
+  integrationTest("DeltaSharingSource.latestOffset - startingVersion latest") {
+    val source = getSource(Map(
+      "ignoreChanges" -> "true",
+      "ignoreDeletes" -> "true",
+      "startingVersion" -> "latest"
+    ))
+    val latestOffset = source.latestOffset(null, source.getDefaultReadLimit)
+    assert(latestOffset == null)
+  }
+
+  integrationTest("DeltaSharingSource.latestOffset - no startingVersion") {
+    val source = getSource(Map("ignoreChanges" -> "true", "ignoreDeletes" -> "true"))
+    val latestOffset = source.latestOffset(null, source.getDefaultReadLimit)
+
+    assert(latestOffset.isInstanceOf[DeltaSharingSourceOffset])
+    val offset = latestOffset.asInstanceOf[DeltaSharingSourceOffset]
+    assert(offset.sourceVersion == 1)
+    assert(offset.tableId == deltaLog.snapshot().metadata.id)
+    assert(offset.tableVersion == 5)
+    assert(offset.index == -1)
+    assert(!offset.isStartingVersion)
+  }
+
+  /**
+   * Test getBatch for Stream CDF
+   */
+  integrationTest("DeltaSharingSource.getBatch - exception") {
+    // getBatch cannot be called without writestream
+    val source = getSource(Map(
+      "ignoreChanges" -> "true",
+      "ignoreDeletes" -> "true",
+      "startingVersion" -> "0"))
+    val latestOffset = source.latestOffset(null, source.getDefaultReadLimit)
+    intercept[AnalysisException] {
+      // Error message would be:
+      //    Queries with streaming sources must be executed with writeStream.start()
+      source.getBatch(None, latestOffset.asInstanceOf[DeltaSharingSourceOffset]).show()
+    }
```

Comment:
Should we test a success cases for getBatch?

Reply:
the success cases of getBatch is tested in all the tests below with the stream query.